modules/rest/src/main/java/eu/cloudnetservice/modules/rest/commands/RestCommand.java

```diff
@@ -0,0 +1,121 @@
+import cloud.commandframework.annotations.CommandMethod;
+import cloud.commandframework.annotations.CommandPermission;
+import eu.cloudnetservice.common.language.I18n;
+import eu.cloudnetservice.driver.CloudNetDriver;
+import eu.cloudnetservice.driver.permission.PermissionUser;
+import eu.cloudnetservice.modules.rest.setup.RestSetup;
+import eu.cloudnetservice.modules.rest.util.ExternalURLBuilder;
+import eu.cloudnetservice.modules.rest.v2.V2HttpHandlerQRCode;
+import eu.cloudnetservice.node.command.source.CommandSource;
+import eu.cloudnetservice.node.command.source.ConsoleCommandSource;
+import eu.cloudnetservice.node.config.ExternalURLInformation;
+import eu.cloudnetservice.node.console.Console;
+import eu.cloudnetservice.node.console.animation.setup.ConsoleSetupAnimation;
+import lombok.NonNull;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class RestCommand {
+
+  private final String header = """
+       __           _  \s
+      /__\\ ___  ___| |_\s
+     / \\/// _ \\/ __| __|
+    / _  \\  __/\\__ \\ |_\s
+    \\/ \\_/\\___||___/\\__|          
+      """;
+  private final String footer = "Rest setup complete!";
+  private final ConsoleSetupAnimation animation = new ConsoleSetupAnimation(this.header, this.footer, "&r> &e");
+  private final Console console;
+
+  private final V2HttpHandlerQRCode v2HttpHandlerQRCode;
+
+  public RestCommand(Console console, V2HttpHandlerQRCode v2HttpHandlerQRCode) {
+    this.console = console;
+    this.v2HttpHandlerQRCode = v2HttpHandlerQRCode;
+  }
+
+  @CommandPermission("cloudnet.command.rest.setup")
+  @CommandDescription("Goes trough the rest setup process")
+  @CommandMethod(value = "rest setup", requiredSender = ConsoleCommandSource.class)
+  public void startSetup(@NonNull CommandSource source) {
+    var setup = new RestSetup(this.v2HttpHandlerQRCode);
+    setup.applyQuestions(this.animation);
+
+    this.animation.addFinishHandler(() -> setup.handleResults(this.animation));
+
+    this.console.startAnimation(this.animation);
+  }
+
+  @CommandDescription("Displays the url of the qrcode")
+  @CommandMethod(value = "rest qrcode")
+  @CommandPermission("cloudnet.command.rest.qrcode")
+  public void showQRCodeUrl(@NonNull CommandSource source) {
+    source.sendMessage(I18n.trans("module-rest-qrcode-info"));
+    this.printExternalUrls(null, source);
+  }
+
+  @CommandDescription("Generates a new qrcode with login information")
+  @CommandMethod(value = "rest login <username> <password>")
+  @CommandPermission("cloudnet.command.rest.qrcode.login")
+  public void loginQRCode(@NonNull CommandSource source, @NonNull @Argument("username") String username,
+    @NonNull @Argument("password") String password) {
+    var users = CloudNetDriver.instance().permissionManagement().usersByName(username);
+    PermissionUser user = null;
+    for (int i = 0; i < users.size() && user == null; i++) {
```

Comment:
use streams

Reply:
I prefer faster primitive loops. Same code at original here but just improved.
<img width="766" alt="image" src="https://user-images.githubusercontent.com/6745190/173209107-01d83ed2-aaee-4cc3-9ddb-d8e1caeea905.png">

Reply:
There are some differences here. As said please use a stream here. Its easier to read & understand